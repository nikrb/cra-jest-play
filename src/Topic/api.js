export const getTopics = () =>
  fetch('/api/v1/topics').then(res => res.json());
