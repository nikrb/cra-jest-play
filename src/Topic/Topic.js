import React from 'react';

import { getTopics } from './api';

export default class Topic extends React.Component {
  state = {
    topics: [],
  };
  componentDidMount = () => {
    getTopics().then(topics => this.setState({ topics }))
  };
  render = () => {
    const topics = this.state.topics.map((topic, ndx) => (
      <li key={ndx}>{topic.name}</li>
    ));
    return (
      <div>
        <ul>
          {topics}
        </ul>
      </div>
    );
  };
};
