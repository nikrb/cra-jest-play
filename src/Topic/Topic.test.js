import React from 'react';
import { shallow } from 'enzyme';

jest.mock('./api');
import Topic from './Topic';

it('should display topic list', () => {
  const wrapper = shallow(<Topic />);
  return Promise.resolve()
    .then(() => {
      wrapper.update();
      expect(wrapper.find('li').length).toBe(2);
      expect(wrapper).toMatchSnapshot();
    });
});
