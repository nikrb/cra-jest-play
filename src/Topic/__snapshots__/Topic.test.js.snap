// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display topic list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topic />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                Topic 1
        </li>
        <li>
                Topics 2
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            Topic 1
</li>,
          <li>
            Topics 2
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Topic 1",
          },
          "ref": null,
          "rendered": "Topic 1",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Topics 2",
          },
          "ref": null,
          "rendered": "Topics 2",
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    Topic 1
          </li>
          <li>
                    Topics 2
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Topic 1
</li>,
            <li>
              Topics 2
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Topic 1",
            },
            "ref": null,
            "rendered": "Topic 1",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Topics 2",
            },
            "ref": null,
            "rendered": "Topics 2",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
